nylas/models/messages.py

```diff
@@ -0,0 +1,179 @@
+ListMessagesQueryParams = TypedDict(
+    "ListMessagesQueryParams",
+    {
+        **get_type_hints(ListQueryParams),  # Inherit fields from ListQueryParams
+        "subject": NotRequired[str],
+        "any_email": NotRequired[str],
+        "from": NotRequired[str],
+        "to": NotRequired[str],
+        "cc": NotRequired[str],
+        "bcc": NotRequired[str],
+        "in": NotRequired[str],
+        "unread": NotRequired[bool],
+        "starred": NotRequired[bool],
+        "thread_id": NotRequired[str],
+        "received_before": NotRequired[int],
+        "received_after": NotRequired[int],
+        "has_attachment": NotRequired[bool],
+        "fields": NotRequired[Fields],
+        "search_query_native": NotRequired[str],
+    },
+)
+"""
+Query parameters for listing messages.
+
+Attributes:
+    subject: Return messages with matching subject.
+    any_email: Return messages that have been sent or received by this comma-separated list of email addresses.
+    from: Return messages sent from this email address.
+    to: Return messages sent to this email address.
+    cc: Return messages cc'd to this email address.
+    bcc: Return messages bcc'd to this email address.
+    in: Return messages in this specific folder or label, specified by ID.
+    unread: Filter messages by unread status.
+    starred: Filter messages by starred status.
+    thread_id: Filter messages by thread_id.
+    received_before: Return messages with received dates before received_before.
+    received_after: Return messages with received dates after received_after.
+    has_attachment: Filter messages by whether they have an attachment.
+    fields: Specify "include_headers" to include headers in the response. "standard" is the default.
+    search_query_native: A native provider search query for Google or Microsoft.
+    limit (NotRequired[int]): The maximum number of objects to return.
+        This field defaults to 50. The maximum allowed value is 200.
+    page_token (NotRequired[str]): An identifier that specifies which page of data to return.
+        This value should be taken from a ListResponse object's next_cursor parameter.
+"""
+
+
+class FindMessageQueryParams(TypedDict):
+
+    """Query parameters for finding a message.
+
+    Attributes:
+        fields: Specify "include_headers" to include headers in the response. "standard" is the default.
+    """
+
+    fields: NotRequired[Fields]
+
+
+class UpdateMessageQueryParams(TypedDict):
+
+    """Query parameters for updating a message.
+
+    Attributes:
+        starred: The message's starred status
+        unread: The message's unread status
+        folders: The message's folders
+    """
+
+    starred: NotRequired[bool]
+    unread: NotRequired[bool]
+    folders: NotRequired[List[str]]
+
+
+class UpdateMessageRequest:
+
+    """Request payload for updating a message."""
+
+    unread: NotRequired[bool]
+    starred: NotRequired[bool]
+    folder: NotRequired[List[str]]
```

Comment:
I don't think there's any query/filter params for updating a message. For the `UpdateMessagRequest` the class hould also have `metadata` as a field that can be update and be a TypedDict:

```suggestion
class UpdateMessageRequest(TypedDict):

    """
    Request payload for updating a message.
    
    Attributes:
        starred: The message's starred status
        unread: The message's unread status
        folders: The message's folders
        metadata: A list of key-value pairs storing additional data
    """

    unread: NotRequired[bool]
    starred: NotRequired[bool]
    folder: NotRequired[List[str]]
    metadata: NotRequired[Dict[str, Any]]
```